### PR TITLE
Escher integration

### PR DESCRIPTION
**Pull request by rohitvarkey**
Escher integration for Vega.

I've added asset files in `assets` folder. We can just do `bower update` to update the dependencies when needed.

This results in interactive features like
```julia
using Vega

function main(window)
    push!(window.assets,("Vega","vega-plot"))
    push!(window.assets,"widgets")
    n = Input(4)
    vbox(
        slider(4:10) >>> n,
        lift(n) do n
            x = 1:n
            y = rand(n)
            barplot(x = x, y = y)
        end
    ) |> pad(2em)
end
```

which would give something like

vega_escher

### CODE REVIEW

**Comment by rohitvarkey** [comment]
Seems like this doesn't do what I was hoping it would. 

```
if Pkg.installed("Escher") != @compat Void 
       include("escher_integration.jl")
end
```

**Comment by rohitvarkey** [comment]
Solved it. `Void` and `nothing` trips me up everytime. :smile:

**Comment by randyzwitch** [comment]
Thanks. Pardon my ignorance, but bower is another package manager? Is that installed when Escher is installed, or is it assumed to be on the user's computer?

**Comment by rohitvarkey** [comment]
Yep! [Bower](http://bower.io/) is a package manager that a lot of web projects use. Polymer (which is what Escher uses) recommends it as well. 

The users need not have bower installed. The updates can be done by a developer who has it installed (which would be you, I guess. :)) It will then take care of updating all the dependencies. And you can add those files and commit like you normally do. Users need not know anything about what's going on.

**Comment by randyzwitch** [comment]
Related #16

**Comment by randyzwitch** [comment]
Unreal, just got it working! Going to hold off until morning to really test out the other functionality to make sure nothing else is affected in the REPL/Jupyter Notebook.

**Comment by rohitvarkey** [comment]
Sure. But I don't think anything will be affected. The new `convert` method will come into play only with Escher. The REPL and Jupyter should be falling back to their `writemime` functions.

**Comment by rohitvarkey** [comment]
I've done a rebase and added a commit removing the extra asset files introduced in #55 and updated paths to the bower components.

**Comment by randyzwitch** [comment]
This appears to keep the REPL working, Jupyter Notebook working and Escher works with your toy example above. Great job!

**Comment by rohitvarkey** [comment]
Thanks! :smile:

**Comment by shashi** [comment]
Great stuff! :) :100:

**Comment by randyzwitch** [comment]
You guys can hammer out the details, I'll just keep working on adding new plot types and mutating functions ;)

File: assets/bower_components/vega-elements/vega-plot.html [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  <script src="../d3-cloud/build/d3.layout.cloud.js"></script>
+  <script src="../vega/vega.min.js"></script>
+
+  <dom-module id="vega-plot">
+    <template>
+      <div id="plot">
+          <content></content>
+      </div>
+      <br>
+      <a href="" id="pnglink" download>Save as PNG</a>
```
Comment (shashi) [inline assets/bower_components/vega-elements/vega-plot.html]: An option to not show this?

**Comment by randyzwitch** [comment]
What do you want to hide?

This is already merged and on METADATA, but if you've got feature requests, feel free to open an issue or submit a PR.

File: assets/bower_components/vega-elements/vega-plot.html (hunk 1 above)
Comment (rohitvarkey) [inline assets/bower_components/vega-elements/vega-plot.html]: I can add that and leave it enabled in Vega.jl .

File: assets/bower_components/vega-elements/vega-plot.html [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            observer: 'render'
+          }
+        },
+        render: function() {
+          var self = this;
+          vg.parse.spec(this.json, function (chart) {
+            chart({el:(Polymer.dom(self.$.plot)).node}).update()
+          });
+          this.async(function (){
+            this.saveAsPNG();
```
Comment (shashi) [inline assets/bower_components/vega-elements/vega-plot.html]: can't this be something like `this.$.pnglink.addEventListener('click', function () { window.location = makeDataUri(); })`?

just doing this on every update seems like an overkill...

Comment (rohitvarkey) [inline assets/bower_components/vega-elements/vega-plot.html]: Yeah. That seems like a good idea.

File: assets/bower_components/vega-elements/vega-plot.html [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<html>
+  <script src="../d3/d3.js"></script>
+  <script src="../topojson/topojson.js"></script>
+  <script src="../d3-cloud/build/d3.layout.cloud.js"></script>
+  <script src="../vega/vega.min.js"></script>
+
+  <dom-module id="vega-plot">
+    <template>
+      <div id="plot">
+          <content></content>
```
Comment (shashi) [inline assets/bower_components/vega-elements/vega-plot.html]: `<content></content>` isn't required right?

Comment (rohitvarkey) [inline assets/bower_components/vega-elements/vega-plot.html]: Just tried removing it. Still works. So I guess not. :)

**Pull request merged**